NewsApp/app/src/main/java/com/thomasvondollen/newsapp/MainActivity.kt

```diff
@@ -66,45 +18,72 @@ class MainActivity : AppCompatActivity() {
         val view = binding.root
         setContentView(view)
 
-        //looping through Article objects and assigning contents to TextViews
-        var count = 0
-
-        for(item in binding.mainGroup.children) {
+        val newsService = NewsService()
+        val newsArticles = newsService.createArticleArray()
 
-            var contents = ""
 
-            val currentArticle = dummyArticles[count]
+        Log.i("MyTag",newsArticles[0].title)
+        Log.i("MyTag",newsArticles[2].author)
 
-            if (item is TextView) {
 
-                //adding content, checking if nullables == null
-                if(currentArticle.source.id != null) {
-                    contents += "${currentArticle.source.id}\n"
-                }
+        newsArticles.forEach { currentArticle ->
+            val articleView = ArticleView(this)
 
-                contents += "${currentArticle.source.name}\n"
+            //articleView.setArticleInfo(currentArticle)
```

Comment:
nit: You can remove this commented code (or the other two method calls below if you'd like).